### PR TITLE
mark language breaking changes in changelog

### PR DESCRIPTION
**Pull request by khieta**
## Description of changes

Proposal to explicitly mark language breaking changes in the CHANGELOG to make them easier for consumers (like [AVP](https://aws.amazon.com/verified-permissions/)) to identify. I'm imagining that we will still _also_ maintain the language version CHANGELOG on [docs.cedarpolicy.com](https://docs.cedarpolicy.com/other/doc-history.html).

### CODE REVIEW

File: cedar-policy/CHANGELOG.md [hunk 1]
```diff
@@ -31,21 +33,21 @@ Cedar Language Version: 4.0
   in 3.2.0. This involved API breaking changes in both. See #757 and #854.
 - Moved `<PolicyId as FromStr>::Err` to `Infallible` (#588, resolving #551)
 - Removed unnecessary lifetimes from some validation related structs (#715)
-- Changed policy validation to reject comparisons and conditionals between
+- (*) Changed policy validation to reject comparisons and conditionals between
   record types that differ in whether an attribute is required or optional. (#769)
-- Changed JSON schema parser so that `Set`, `Entity`, `Record`, and `Extension`
+- (*) Changed JSON schema parser so that `Set`, `Entity`, `Record`, and `Extension`
```
Comment (khieta) [inline cedar-policy/CHANGELOG.md:38]: Note that we plan to undo this change (#1070), but noting it anyways for consistency.

**Review by john-h-kastner-aws — APPROVED** [review]
This way of labeling changes looks good to me

**Review by shaobo-he-aws — APPROVED** [review]
